### ISSUE TITLE
NVDA crashes in windows 10 event viewer

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#9222]
### Steps to reproduce:
1. Open NVDA
2. In Windows 10 1809 update, Open the event viewer
3. In the tree view choose a cathegory (i.e. applications)
4. Press tab to land into the list with events
5. Press enter on an event to open its properties
6. Press tab to read the even
7. Press shift+tab until you land on "close button"
8. Press space bar

### Actual behavior:
Often NVDA crashes. You must start NVDA again to solve the pbroblem. After restarting NVDA, the event viewer window cannot be accessed. You hear a tone.

### Expected behavior:
NVDA should not crash and the event viewer window should be accessible again.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2018.4.1

#### Windows version:
Windows 10 1809 update

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA?
no

**Comment by Adriani90** [comment]
I get following in the log file:

```
Input: kb(laptop):shift+tab
IO - speech.speak (18:11:10.247):
Speaking [u'Allgemein  Registerkarte  Ausgew\xe4hlt']
IO - inputCore.InputManager.executeGesture (18:11:10.437):
Input: kb(laptop):shift+tab
IO - speech.speak (18:11:10.497):
Speaking [u'Schlie\xdfen  Schalter']
IO - inputCore.InputManager.executeGesture (18:11:11.017):
Input: kb(laptop):space
IO - speech.speak (18:11:11.038):
Speaking [u'Gedr\xfcckt']
DEBUGWARNING - watchdog._watcher (18:11:11.618):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 217, in <module>
  File "core.pyo", line 517, in main
  File "wx\core.pyo", line 2134, in MainLoop
  File "gui\__init__.pyo", line 963, in Notify
  File "core.pyo", line 488, in run
  File "queueHandler.pyo", line 86, in pumpAll
  File "queueHandler.pyo", line 53, in flushQueue
  File "eventHandler.pyo", line 62, in _queueEventCallback
  File "eventHandler.pyo", line 150, in executeEvent
  File "comtypesMonkeyPatches.pyo", line 54, in newCpbDel
  File "comtypes\__init__.pyo", line 919, in __del__
  File "comtypes\__init__.pyo", line 1174, in Release

IO - speech.speak (18:11:12.217):
Speaking [u'Fenster']
DEBUGWARNING - watchdog._watcher (18:11:12.737):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 217, in <module>
  File "core.pyo", line 517, in main
  File "wx\core.pyo", line 2134, in MainLoop
  File "gui\__init__.pyo", line 963, in Notify
  File "core.pyo", line 487, in run
  File "IAccessibleHandler.pyo", line 900, in pumpAll
  File "IAccessibleHandler.pyo", line 657, in processFocusWinEvent
  File "IAccessibleHandler.pyo", line 532, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible\__init__.pyo", line 40, in getNVDAObjectFromEvent
  File "IAccessibleHandler.pyo", line 342, in accessibleObjectFromEvent
  File "oleacc.pyo", line 265, in AccessibleObjectFromEvent

DEBUGWARNING - core.CorePump.run (18:11:12.848):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyo", line 487, in run
  File "IAccessibleHandler.pyo", line 900, in pumpAll
  File "IAccessibleHandler.pyo", line 657, in processFocusWinEvent
  File "IAccessibleHandler.pyo", line 532, in winEventToNVDAEvent
  File "NVDAObjects\IAccessible\__init__.pyo", line 46, in getNVDAObjectFromEvent
  File "NVDAObjects\__init__.pyo", line 72, in __call__
  File "watchdog.pyo", line 202, in _COMError_init
CallCancelled
IO - inputCore.InputManager.executeGesture (18:11:13.997):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (18:11:15.298):
Input: kb(laptop):shift+tab
IO - inputCore.InputManager.executeGesture (18:11:15.865):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (18:11:16.987):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (18:11:19.167):
Input: kb(laptop):e
IO - inputCore.InputManager.executeGesture (18:11:19.322):
Input: kb(laptop):r
IO - inputCore.InputManager.executeGesture (18:11:19.822):
Input: kb(laptop):enter
IO - inputCore.InputManager.executeGesture (18:11:24.368):
Input: kb(laptop):windows+m
IO - inputCore.InputManager.executeGesture (18:11:24.546):
Input: kb(laptop):windows+m
IO - inputCore.InputManager.executeGesture (18:11:25.746):
Input: kb(laptop):ö
IO - inputCore.InputManager.executeGesture (18:11:26.517):
Input: kb(laptop):n
IO - inputCore.InputManager.executeGesture (18:11:26.647):
Input: kb(laptop):v
IO - inputCore.InputManager.executeGesture (18:11:27.247):
Input: kb(laptop):enter
IO - inputCore.InputManager.executeGesture (18:11:42.407):
Input: kb(laptop):windows+m
IO - inputCore.InputManager.executeGesture (18:11:43.677):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (18:11:46.118):
Input: kb(laptop):upArrow
IO - inputCore.InputManager.executeGesture (18:11:46.276):
Input: kb(laptop):upArrow
IO - inputCore.InputManager.executeGesture (18:11:47.038):
Input: kb(laptop):windows+m
IO - inputCore.InputManager.executeGesture (18:11:47.197):
Input: kb(laptop):windows+m
IO - inputCore.InputManager.executeGesture (18:11:47.516):
Input: kb(laptop):ö
IO - inputCore.InputManager.executeGesture (18:11:48.052):
Input: kb(laptop):upArrow
```

**Comment by Adriani90** [comment]
Tested with 2018.3.2, the crash is not occuring.

I get following error in event viewer window, the error dialog came up suddenly after trying to access the window after crashing:

```
FX:{b05566ad-fe9c-4363-be05-7a4cbb7cb510}
The object reference was not defined for an object instance.
System.NullReferenceException
   at Microsoft.ManagementConsole.Executive.SnapInNotificationOperation.ProcessNotification()
   at Microsoft.ManagementConsole.Executive.Operation.OnThreadTransfer(SimpleOperationCallback callback)
```

**Comment by Adriani90** [comment]
cc: @feerrenrut, @michaelDCurran

**Comment by tspivey** [comment]
I managed to get this to crash. However, when it crashed, something I did caused the keyboard hook to freeze, just like in #4669.

NVDA should never freeze the entire keyboard. Once it does that, there's no way back to a working system.

**Comment by Adriani90** [comment]
Still reproducible in NVDA 2019.3 Beta 1.
cc: @feerrenrut which priority would you assign to this? I my view it is a quite severe issue since the keyboard does not work for quite some time and switching windows does not work either.

**Comment by Adriani90** [comment]
I cannot reproduce this with NVDA 2022.4 and Windows 11. @tspivey are you able to test this under Windows 10 and see if it still occuring? Otherwise I would tend to close this issue since I think not many users will encounter this one in particular.

**Comment by tspivey** [comment]
When I try the steps in the description, Event Viewer stops responding. NVDA stays up, but Alt+Tab no longer reads until I close the window.
